test/local/lib/senders/index.js

```diff
@@ -0,0 +1,284 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const ROOT_DIR = '../../../..'
+
+const assert = require('insist')
+const config = require(`${ROOT_DIR}/config`).getProperties()
+const crypto = require('crypto')
+const error = require(`${ROOT_DIR}/lib/error`)
+const senders = require(`${ROOT_DIR}/lib/senders`)
+const sinon = require('sinon')
+const P = require('bluebird')
+
+const nullLog = {
+  trace() {},
+  info() {},
+  error() {}
+}
+
+describe('lib/senders/index', () => {
+
+  describe('email', () => {
+
+    const UID = crypto.randomBytes(16)
+    const EMAIL = crypto.randomBytes(16).toString('hex') + '@example.test'
+    const db = {
+      emailBounces: sinon.spy(() => P.resolve([]))
+    }
+
+    function createSender(config, db) {
+      return senders(config, nullLog, error, db)
+        .then(sndrs => {
+          const email = sndrs.email
+          email._ungatedMailer.mailer.sendMail = sinon.spy((opts, cb) => {
+            cb(null, {})
+          })
+          return email
+        })
+    }
+
+    beforeEach(() => {
+      db.emailBounces.reset()
+    })
+
+    describe('.sendVerifyCode()', () => {
+      const acct = {
+        email: EMAIL,
+        uid: UID
+      }
+      const code = crypto.randomBytes(32)
+
+      it('should call mailer.verifyEmail()', () => {
+        let email
+        return createSender(config, db)
+          .then(e => {
+            email = e
+            email._ungatedMailer.verifyEmail = sinon.spy(() => P.resolve({}))
+            return email.sendVerifyCode(acct, code, {})
+          })
+          .then(() => {
+            assert.equal(db.emailBounces.callCount, 1)
+            assert.equal(email._ungatedMailer.verifyEmail.callCount, 1)
+          })
+      })
+    })
+
+    describe('.sendVerifyLoginEmail()', () => {
```

Comment:
There seems to be some test boilerplate/duplication that could be replaced with some dynamic testing. For example, could use the following to test tests that use tokens.

```javascript
    var sendVerifyCodeTest = [
      {
        emailFunc: 'sendVerifyCode',
        mailerFunc: 'verifyEmail'
      }, {
        emailFunc: 'sendVerifyLoginEmail',
        mailerFunc: 'verifyLoginEmail'
      }
    ]
    sendVerifyCodeTest.forEach(function (testCase) {
      describe(testCase.emailFunc + '()', () => {
        const acct = {
          email: EMAIL,
          uid: UID
        }
        const code = crypto.randomBytes(32)

        it('should call mailer.' + testCase.mailerFunc + '()', () => {
          let email
          return createSender(config, db)
            .then(e => {
              email = e
              email._ungatedMailer[testCase.mailerFunc] = sinon.spy(() => P.resolve({}))
              return email[testCase.emailFunc](acct, code, {})
            })
            .then(() => {
              assert.equal(db.emailBounces.callCount, 1)
              assert.equal(email._ungatedMailer[testCase.mailerFunc].callCount, 1)
            })
        })
      })
    })
```

Not sure if you are pro/against this kind of testing though.